### PR TITLE
Pipe publishing parameters for v3

### LINKED ISSUE #6514: Publishing v3 doesn't pass addintional parameters
- [ ] This issue is blocking 
- [x] This issue is causing unreasonable pain

`symbolPublishingAdditionalParameters` are not getting passed to the the PS1 scripts so darc won't add the build appropriately. This caused a failure that got swallowed silently because of #6403

### PR DESCRIPTION
**Pull request by hoyosjs**
Fixes #6514

### CODE REVIEW

**Comment by hoyosjs** [comment]
@riarenas I am just preserving functionality from v2. darc, the ps script, and the yml already had such parameters.

File: eng/common/templates/post-build/post-build.yml [hunk 1]
```diff
@@ -266,6 +266,8 @@ stages:
               -MaestroToken '$(MaestroApiAccessToken)'
               -WaitPublishingFinish ${{ parameters.waitPublishingFinish }}
               -PublishInstallersAndChecksums ${{ parameters.publishInstallersAndChecksums }}
+              -ArtifactsPublishingAdditionalParameters ${{ parameters.artifactsPublishingAdditionalParameters }}
```
Comment (riarenas) [inline eng/common/templates/post-build/post-build.yml:269]: What exactly do we require still to pass here? The change looks harmless enough, but I'd honestly like these two parameters to not exist at all.

**Comment by riarenas** [comment]
>I am just preserving functionality from v2. darc, the ps script, and the yml already had such parameters.

I'm mostly wondering if this is the moment to try and get rid of them as we're bringing in a new version of the publishing infra, instead of preserving them.

File: eng/common/templates/post-build/post-build.yml (hunk 1 above)
Comment (hoyosjs) [reply eng/common/templates/post-build/post-build.yml:269]: For the symbols we need it as there are some changes in the symbol uploader that matter only to coreclr (runtime) and to diagnostics. They change special indices necessary for managed debuggers.

Comment (riarenas) [reply eng/common/templates/post-build/post-build.yml:269]: That makes sense, and the fact that the script already takes these parameters makes me think this was just an oversight and not a design decision for v3. I'll approve but I think @epananth is the best person to confirm we want these passed through.

**Review by riarenas — APPROVED** [review]

**Comment by hoyosjs** [comment]
:( I manually called  `darc add-build-to-channel` with `--symbol-publishing-parameters` and it didn't get piped down, although I can see the pipeline did receive the value. I'll update the pipeline.

File: eng/publishing/v3/publish.yml [hunk 2]
```diff
@@ -56,7 +56,7 @@ stages:
         
     - template: /eng/publishing/v3/publish-symbols.yml
       parameters:
-        publishInstallersAndChecksums: ${{ parameters.publishInstallersAndChecksums }}
+        symbolPublishingAdditionalParameters: ${{ parameters.symbolPublishingAdditionalParameters }}
```
Comment (epananth) [inline eng/publishing/v3/publish.yml:59]: im not sure why you removed publishInstallerAndChecksums this is a required attribute for add build-to-channel

Comment (hoyosjs) [reply eng/publishing/v3/publish.yml:59]: O only removed it from publish-symbols. This one doesn't have such a parameter.

**Comment by hoyosjs** [comment]
@epananth if this looks good, feel free to merge.

**Comment by epananth** [comment]
@hoyosjs did you queue any test build to make sure this worked?

**Comment by hoyosjs** [comment]
@epananth i don't know how to test the promotion pipeline. Is there a way to run it off a non master branch?

**Comment by epananth** [comment]
yes you can, in darc add-build-to-channel use --source-branch < test branch instead>. I will connect with you on teams..

**Review by epananth — APPROVED** [review]

**Pull request merged**